### PR TITLE
Migrating Auth API to the new Identity Toolkit endpoint

### PR DESCRIPTION
**Pull request by hardikns**
Using the new Identity Toolkit REST endpoint for user management features. This is as per corresponding change in java (firebase/firebase-admin-java#220) and node SDKs.

### CODE REVIEW

**Comment by hardikns** [comment]
@hiranya911 Can you please help on how to run Integration tests? I would like to do those before I submit the PR.

**Comment by hiranya911** [comment]
@hardikns Check the CONTRIBUTING.md file for instructions.

**Comment by hardikns** [comment]
@hiranya911 Can you please review the change? I plan to submit another in few days with changes around ooblink. 

Meanwhile I did not find any documentation of the new Identity Toolkit.  If you can point to the same that would help.

**Comment by hiranya911** [comment]
@hardikns thanks for doing this. I will review this sometime this week.

@bojeil-google any pointers to the documentation of the new ID toolkit API?

File: tests/test_user_mgt.py [hunk 1]
```diff
@@ -26,6 +26,11 @@
 from firebase_admin import _user_mgt
 from tests import testutils
 
+try:
```
Comment (hiranya911) [inline tests/test_user_mgt.py]: Import via `six`:

```
from six.moves.urllib import parse

parse.urlsplit()
parse.parse_qsl()
```

Comment (hardikns) [reply tests/test_user_mgt.py]: This one gives pylint issue

```
$ ./lint.sh all
Running linter on module firebase_admin
Running linter on module tests
************* Module tests.test_user_mgt
E: 29, 0: Unable to import 'six.moves.urllib' (import-error)
```

Do you think upgrading pylint will solve issue?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # Unreleased
 
+- [added] Migrated the `FirebaseAuth` user management API to the
```
[the remark this answers is not in the document]
Comment (hardikns) [reply CHANGELOG.md]: fixed.

File: firebase_admin/auth.py [hunk 3]
```diff
@@ -466,13 +466,20 @@ def __init__(self, code, message, error=None):
...
+    ID_TOOLKIT_URL = 'https://identitytoolkit.googleapis.com/v1/projects/'
 
     def __init__(self, app):
         credential = app.credential.get_credential()
         version_header = 'Python/Admin/{0}'.format(firebase_admin.__version__)
+
+        if not app.project_id:
+            raise ValueError("""Project ID is required to access the auth service.
+            1. Use a service account credential, or
+            2. set the project ID explicitly via FirebaseOptions, or
```
[the remark this answers is not in the document]
Comment (hardikns) [reply firebase_admin/auth.py]: fixed

File: tests/test_user_mgt.py (hunk 1 above)
Comment (hardikns) [reply tests/test_user_mgt.py]: fixed

**Comment by hardikns** [comment]
@bojeil-google can you please point me to the documentation on the new ID toolkit API?

**Comment by bojeil-google** [comment]
If you mean the REST API, it is not yet documented.

**Pull request merged**